extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -161,46 +251,35 @@ public function csvStats() {
 
 		header('Content-Disposition: attachment; filename="' . $name . '.csv"');
 
-		echo implode( ",", $this->statsHeaders ) . "\n";
+		echo implode( ",", self::STATS_HEADERS ) . "\n";
 
 		foreach ( $stats['data'] as $dataRow ) {
 			echo implode( ",", $dataRow ) . "\n";
 		}
 
-		exit;
+		return false;
 	}
 
-	protected function getPageTitle() {
-		return 'Image Review tool';
-	}
-
-	protected function getHelper() {
-		return new ImageReviewHelper();
-	}
-
-	protected function getBaseUrl() {
-		return Title::newFromText('ImageReview', NS_SPECIAL)->getFullURL();
-	}
-
-	protected function getToolName() {
-		return 'Image Review';
-	}
+	protected function getStatsData( $startYear, $startMonth, $startDay,
+	                                 $endYear, $endMonth, $endDay ) {
 
-	protected function getStatsPageTitle() {
-		return 'Image Review tool statistics';
+		return $this->statsDataGetter->getStatsData(
+			$startYear, $startMonth, $startDay,
+			$endYear, $endMonth, $endDay
+		);
 	}
 
 	protected function parseImageData($data) {
-		$images = array();
+		$images = [];
 
 		foreach( $data as $name => $value ) {
-			if (preg_match('/img-(\d*)-(\d*)/', $name, $matches)) {
+			if ( preg_match( '/img-(\d*)-(\d*)/', $name, $matches ) ) {
 				if ( !empty($matches[1]) && !empty($matches[2]) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)